url65a69497a4_sweider_BankSystem_JavaLab8_tgz-pJ8-banksystem_lab4_LauncherJ8/src/banksystem/lab4/client/Client.java

```diff
@@ -0,0 +1,128 @@
+        }
+        
+        cashier.releaseCashier(this.id);
+        logger.info("Cashier released");
+        
+        logger.info("Finish invokation");
+    }
+
+    private boolean transferMoney(Random rand) {
+        int value = rand.nextInt();
+        MoneyAmount moneyToTransfer = new MoneyAmount(value);
+        
+        Cashier.TransferMoneyToAccountOperationResult result;
+         
+        do{
+            result = cashier.transferMoneyToAccount(this.id,this.accounts.iterator().next() ,
+                    moneyToTransfer);
+            
+            switch (result) {
+                case ACCOUNT_NOT_SETTED: {
+                    logger.warn("Account not setted!");
+                    return false;
+                } 
+                
+                case CASHIER_IS_NOT_OCCUPIED: logger.warn("Cashier isn't occupied");return false;
+                case CASHIER_IS_BUSY: logger.warn("Cashier work with other."); return false;
+                case INVALID_RECIEVER_ID: logger.warn("Invalid reciever id"); return false;
+                case TRANSACTION_CREATED: logger.info("Successfully created transaction about " + value +"$.");break;
+                case NOT_ENOUGH_MONEY: {
+                    logger.info("Not enough money to transfer "+value+"$. Trying other value.");
+                    value = rand.nextInt();
+                    moneyToTransfer = new MoneyAmount(value);
+                } break; 
+            }
+     
+        }while(result != Cashier.TransferMoneyToAccountOperationResult.TRANSACTION_CREATED);
+        return true;
+    }
+
+    private boolean startWorkWithAccount(Integer accountId) {
+        logger.debug("Try to start work with account#" + accountId);
+        Cashier.StartWorkWithAccountOperationResult result = cashier.startWorkWithAccount(this.id, accountId);
+        if (result == Cashier.StartWorkWithAccountOperationResult.ACCOUNT_NOT_FOUND) {
+            logger.warn("Account#" + accountId + " was not found");
+            return false;
+        }
+        
+        while(result != Cashier.StartWorkWithAccountOperationResult.SUCCESSFULLY_STARTED){
+            switch (result) {
+                case CASHIER_IS_BUSY: logger.warn("Cashier work with other");break;
+                case CASHIER_IS_NOT_OCCUPIED: logger.info("Cashier is not occupied"); break;
+
+                default:  {
+                    logger.warn("Other case: " + result);
+                    this.occupyCashier();
+                }
+            }
+            result = cashier.startWorkWithAccount(this.id, accountId);            
+        }
+        return true;
+    }
+
+    private void occupyCashier() {
+        boolean occupied = cashier.occupyCashier(this.id);
+        
+        while(occupied == false){
+            logger.info("Cashier is busy. Waiting.");
+            try { Thread.sleep(200); } catch (InterruptedException ex) { logger.error(null, ex); }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)